### DISCUSSION TITLE
how to use time::interval and tick.

### DISCUSSION THREAD

**Discussion by luckychacha** [tokio-rs/tokio#4216] [Q&A] [answered]
Hello,I am a new comer to learn async and tokio. There is a question I don't know how to deal with.
Here is the code. I want the output is : Every 10 seconds print "dancing",
but now the output is print "dancing" all the time, seems that `i.tick().await;` not worked.
How can I fix it. thanks.

```

use std::time::Duration;

use chrono::Local;
use tokio::time;

async fn dance() {
    let mut i = time::interval(Duration::from_secs(10));
    loop {
        i.tick().await;
        {
            print_time();

            println!("dancing");
        }
    }
}

async fn async_main() {
        tokio::spawn(async move {
            dance().await;
        });
}

#[tokio::main]
async fn main() {
    async_main().await;
}

fn print_time() {
    let fmt = "%Y-%m-%d %H:%M:%S";
    println!("current time is : {}", Local::now().format(fmt).to_string());
}

```

**Comment by sfackler** [comment, marked as answer]
You are creating an infinite amount of `dance` tasks that are all running concurrently.

**Reply by luckychacha** [reply]
thanks, I remove the `loop` in async_main and add an `await` after `tokio::spawn` and it works.
